common/plugins/org.jboss.tools.common.core/src/org/jboss/tools/common/core/jandex/JandexUtil.java

```diff
@@ -0,0 +1,71 @@
+/******************************************************************************* 
+ * Copyright (c) 2015 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.common.core.jandex;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Enumeration;
+import java.util.StringTokenizer;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Index;
+import org.jboss.jandex.Indexer;
+import org.jboss.tools.common.core.CommonCorePlugin;
+
+public class JandexUtil {
+
+	public static Index createJarIndex(File jarFile, Indexer indexer) throws IOException {
```

Comment:
Ok, I did to new PR:  https://github.com/jbosstools/jbosstools-base/pull/414, https://github.com/jbosstools/jbosstools-javaee/pull/346 that incapsulate Jandex inside common and open two query methods that are used by Batch and CDI.